frontend/src/stimulus/controllers/dynamic/members-form.controller.ts

```diff
@@ -0,0 +1,142 @@
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { Controller } from '@hotwired/stimulus';
+
+export default class MembersFormController extends Controller {
+  static targets = [
+    'filterContainer',
+    'filterMemberButton',
+    'addMemberForm',
+    'search',
+    'memberUserIds',
+    'addMemberButton',
+    'membershipEditForm',
+    'errorExplanation',
+    'limitWarning',
+  ];
+
+  declare readonly filterContainerTarget:HTMLElement;
+
+  declare readonly filterMemberButtonTarget:HTMLButtonElement;
+
+  declare readonly addMemberFormTarget:HTMLElement;
+
+  declare readonly memberUserIdsTarget:HTMLInputElement;
+
+  declare readonly addMemberButtonTarget:HTMLButtonElement;
+
+  declare readonly membershipEditFormTargets:HTMLElement[];
+
+  declare readonly errorExplanationTarget:HTMLElement;
+
+  declare readonly hasErrorExplanationTarget:HTMLElement;
+
+  declare readonly limitWarningTarget:HTMLElement;
+
+  connect() {
+    // Show/Hide content when page is loaded
+    if (window.OpenProject.guardedLocalStorage('showFilter') === 'true') {
+      this.showFilter();
+    } else {
+      this.hideFilter();
+      // In case showFilter is not set yet
+      window.OpenProject.guardedLocalStorage('showFilter', 'false');
+    }
+
+    if (this.hasErrorExplanationTarget && this.errorExplanationTarget.textContent !== '') {
+      this.showAddMemberForm();
+    }
+
+    if (this.addMemberButtonTarget.getAttribute('data-trigger-initially')) {
+      this.showAddMemberForm();
+    }
+  }
+
+  hideFilter() {
+    this.filterContainerTarget.classList.add('collapsed');
+  }
+
+  showFilter() {
+    this.filterContainerTarget.classList.remove('collapsed');
+  }
+
+  hideAddMemberForm() {
+    this.addMemberFormTarget.style.display = 'none';
+    this.addMemberButtonTarget.focus();
+    this.addMemberButtonTarget.removeAttribute('disabled');
+  }
+
+  showAddMemberForm() {
+    this.addMemberFormTarget.style.display = 'block';
+    this.hideFilter();
+    this.filterMemberButtonTarget.classList.remove('-active');
+    window.OpenProject.guardedLocalStorage('showFilter', 'false');
+    this.addMemberButtonTarget.setAttribute('disabled', 'true');
+
+    const select = this.addMemberFormTarget.querySelector<HTMLInputElement>('.ng-input input');
+    select?.focus();
+
+    this.memberUserIdsTarget.addEventListener('mouseover', () => {
```

Comment:
This event listener never gets removed, prefer to use the `data-action="mouseover->members-form#methodCall`

Reply:
Thank you, still getting used to it :)